File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        int open = 0;
        string ans = "";
        for(int i=0; i<s.size(); i++) {
            if(s[i] == '(') {
                open++;
                if(open > 1) ans += s[i];
            }
            else {
                open--;
                if(open > 0) ans += s[i];
            }
        }
        return ans;
    }
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int open = 0, n = s.length();
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<n; i++) {
            if(s.charAt(i) == '(') {
                if(open > 0) sb.append(s.charAt(i));
                open++;
            }
            else {
                if(open > 1) sb.append(s.charAt(i));
                open--;
            }
        }
        return sb.toString();
    }
}